src/sentry/migrations/0203_release_adoption.py

```diff
@@ -0,0 +1,139 @@
+                    ALTER TABLE "sentry_release_project" ADD COLUMN "unadopted" timestamp with time zone NULL;
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "sentry_release_project" DROP COLUMN "unadopted";
+                    """,
+                ),
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_releaseprojectenvironment" ADD COLUMN "adopted" timestamp with time zone NULL;
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "sentry_releaseprojectenvironment" DROP COLUMN "adopted";
+                    """,
+                ),
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_releaseprojectenvironment" ADD COLUMN "unadopted" timestamp with time zone NULL;
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "sentry_releaseprojectenvironment" DROP COLUMN "unadopted";
+                    """,
+                ),
+                migrations.RunSQL(
+                    """
+                    CREATE INDEX CONCURRENTLY IF NOT EXISTS "sentry_release_project_proj_id_adopted_4ce765fa" ON "sentry_release_project" ("project_id", "adopted");
+                    """,
+                    reverse_sql="DROP INDEX CONCURRENTLY IF EXISTS sentry_release_project_proj_id_adopted_4ce765fa",
+                ),
+                migrations.RunSQL(
+                    """
+                    CREATE INDEX CONCURRENTLY IF NOT EXISTS "sentry_release_project_proj_id_unadopted_8h5g84ee" ON "sentry_release_project" ("project_id", "unadopted");
+                    """,
+                    reverse_sql="DROP INDEX CONCURRENTLY IF EXISTS sentry_release_project_proj_id_unadopted_8h5g84ee",
+                ),
+                migrations.RunSQL(
+                    """
+                    CREATE INDEX CONCURRENTLY IF NOT EXISTS "sentry_releaseprojectenvironment_proj_id_adopted_cf1f902b" ON "sentry_releaseprojectenvironment" ("project_id", "adopted");
+                    """,
+                    reverse_sql="DROP INDEX CONCURRENTLY IF EXISTS sentry_releaseprojectenvironment_proj_id_adopted_cf1f902b",
+                ),
+                migrations.RunSQL(
+                    """
+                    CREATE INDEX CONCURRENTLY IF NOT EXISTS "sentry_releaseprojectenvironment_proj_id_unadopted_bac2355c" ON "sentry_releaseprojectenvironment" ("project_id", "unadopted");
+                    """,
+                    reverse_sql="DROP INDEX CONCURRENTLY IF EXISTS sentry_releaseprojectenvironment_proj_id_unadopted_bac2355c",
+                ),
+                migrations.RunSQL(
+                    """
+                    CREATE INDEX CONCURRENTLY IF NOT EXISTS "sentry_releaseprojectenvironment_proj_id_env_id_adopted_j6h89s3f" ON "sentry_releaseprojectenvironment" ("project_id", "environment_id", "adopted");
```

Comment:
The cols in this look different to the definition on the model - should be `("project_id", "adopted", "environment_id")`

Reply:
Oh damn, I didn't realize they had to be in the same order. So would this index just not have been used? And I guess Django would not have detected that

Reply:
Now I'm wondering how the order affects the index's performance. Could a different order here vastly affect the speed of the query?

Reply:
It matters for different types of queries.

So for an index like `"project_id", "environment_id", "adopted"`, queries like `project_id=1`, `project_id=1 and environment_id=2` and `project_id=2 and environment_id=2 and adopted=True` all work well. `project_id=1 and adopted=True` will still use the index, but it's not as fast since we have to scan through all environments to see whether the rows are adopted or not.

For an index like ``"project_id", "adopted", "environment_id"` queries like `project_id=1`, `project_id=1 and adopted=True` and `project_id=2 and environment_id=2 and adopted=True` are all fine. I think we're more likely to make queries that follow this pattern.

Reply:
Aah okay, that makes sense, thanks for the explanation!